packages/table-core/src/utils/getGroupedRowModel.ts

```diff
@@ -176,7 +176,7 @@ function groupBy<TData extends RowData>(rows: Row<TData>[], columnId: string) {
     if (!previous) {
       map.set(resKey, [row])
     } else {
-      map.set(resKey, [...previous, row])
```

Comment:
In cases of large clusters of values, this operation gets more and more expensive every iteration - recreating 49k entries, recreating 49,001 entries, recreating 49,002 entries, etc